# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/CanvasDrawEventRecorder.h
# Commit: de6ddf5eb2a7
# Full Hash: de6ddf5eb2a7bd3d8a2f58636760e0aa16c584b9
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 773e7b31ac6e -r de6ddf5eb2a7 gfx/layers/CanvasDrawEventRecorder.h
--- a/gfx/layers/CanvasDrawEventRecorder.h	Tue Dec 05 09:37:03 2023 +0000
+++ b/gfx/layers/CanvasDrawEventRecorder.h	Tue Dec 05 09:37:03 2023 +0000
@@ -177,7 +177,7 @@
    * @param aData the data to be written back to the writer
    * @param aSize the number of chars to write
    */
-  void ReturnWrite(const char* aData, size_t aSize);
+  void ReturnWrite(const uint8_t* aData, size_t aSize);
 
   /**
    * Used to read data sent back from the reader via ReturnWrite. This is done
@@ -187,7 +187,11 @@
    * @param aOut the pointer to read into
    * @param aSize the number of chars to read
    */
-  void ReturnRead(char* aOut, size_t aSize);
+  void ReturnRead(uint8_t* aOut, const gfx::IntSize& aSize, size_t aBPP,
+                  size_t aStride = 0);
+  void ReturnRead(uint8_t* aOut, size_t aSize) {
+    ReturnRead(aOut, gfx::IntSize(aSize, 1), 1);
+  }
 
   bool UsingLargeStream() { return mLargeStream; }
 
@@ -290,7 +294,12 @@
 
   void Flush() final {}
 
-  void ReturnRead(char* aOut, size_t aSize) {
+  void ReturnRead(uint8_t* aOut, const gfx::IntSize& aSize, size_t aBPP,
+                  size_t aStride = 0) {
+    mOutputStream.ReturnRead(aOut, aSize, aBPP, aStride);
+  }
+
+  void ReturnRead(uint8_t* aOut, size_t aSize) {
     mOutputStream.ReturnRead(aOut, aSize);
   }
 